### PR TITLE
catch exceptions in handler function and wait until handler completes

### PR DESCRIPTION
**Pull request by karthik2804**
This PR adds a `try...catch` to the template so that any unhandled exceptions are caught and returned as an error instead of the generic error along the lines of `event loop error - both task and job queues are empty, but expected operations did not resolveInternal error.`

This PR also modifies the template so that there is an `event.waitUntill` on the handler function so that applications can do some work even after the response is sent.

### CODE REVIEW

**Comment by karthik2804** [comment]
@tschneidereit I have made the change as discussed.

File: templates/http-ts/content/src/spin.ts [hunk 1]
```diff
@@ -18,7 +18,16 @@ async function handleEvent(event: FetchEvent) {
 
     let res = new ResponseBuilder(resolve);
 
-    await handler(event.request, res)
+    try {
+        // In case you want to do sonme work after the response is sent
+        // uncomment the line below and comment out the line with 
+        // `await handler(event.request, res)
+        // event.waitUntil(handler(event.request, res))
```
Comment (karthik2804) [inline templates/http-ts/content/src/spin.ts]: happy to replace this with a boolean and `if...else` statement if that would be cleaner.

File: templates/http-ts/content/src/spin.ts [hunk 2]
```diff
@@ -18,7 +18,16 @@ async function handleEvent(event: FetchEvent) {
 
     let res = new ResponseBuilder(resolve);
 
-    await handler(event.request, res)
+    try {
+        // In case you want to do sonme work after the response is sent
```
Comment (radu-matei) [inline templates/http-ts/content/src/spin.ts]: typo

File: templates/http-ts/content/src/spin.ts [hunk 3]
```diff
@@ -18,7 +18,16 @@ async function handleEvent(event: FetchEvent) {
...
+        // uncomment the line below and comment out the line with 
+        // `await handler(event.request, res)
```
Comment (radu-matei) [inline templates/http-ts/content/src/spin.ts]: extra '`' symbol here

File: templates/http-ts/content/src/spin.ts [hunk 4]
```diff
@@ -18,7 +18,16 @@ async function handleEvent(event: FetchEvent) {
...
+        // event.waitUntil(handler(event.request, res))
+        await handler(event.request, res)
+    } catch (e: any) {
+        res.status(500)
+        res.send(`error in handler: ${e}`)
+    }
 }
 
 // Keep wizer happy during pre-init. Should go away
```
Comment (radu-matei) [inline templates/http-ts/content/src/spin.ts]: missing space on the line below

Comment (radu-matei) [reply templates/http-ts/content/src/spin.ts]: The issue linked here is closed -- is this still needed?

Comment (karthik2804) [reply templates/http-ts/content/src/spin.ts]: I will clean it up. Thoughts on the comment vs using a `if...else` if you want to continue processing after response has been sent back?

Comment (radu-matei) [reply templates/http-ts/content/src/spin.ts]: I don't really have a preference.

Comment (karthik2804) [reply templates/http-ts/content/src/spin.ts]: leaving it as  a comment, since it is a template, we can change it if we feel the need for it.

**Pull request merged**